src/components/widgets/Form.js

```diff
@@ -0,0 +1,315 @@
+          this.setState({
+            [ nestedDropdownValue.name ]: nestedDropdownValue.defaultValue
+          })
+        }
+      })
+    })
+  }
+
+  setFieldState( state, stateValue ) {
+    this.setState({
+      [ state ]: stateValue
+    })
+  }
+
+  submitWidget( state ) {
+    const widgetData = {}
+    for ( let key in state ) {
+      widgetData[key] = state[key]
+      widgetData.y = '10px'
+      widgetData.x = '20px'
+    }
+    console.log('You submitted a widget with this data:', widgetData)
+  }
+
+  render() {
+    const { type, sections, description, image } = this.props
+
+    return (
+      <div className="widget-form-container">
+        <div className="widget-form-header">
+          <h1>{type}</h1>
+        </div>
+        <form
+          onSubmit={() => this.submitWidget( this.state )}
+          className="widget-form">
+          {sections.map(( section, id ) => (
+              <Section
+                {...section}
+                setFieldState={this.setFieldState}
+                state={this.state}
+                key={`section-${id}`}
+              />
+            )
+          )}
+          <div className="widget-form-buttons">
+            <Button content="Add Widget" type="submit"/>
+            <Button content="Cancel" type="cancel"/>
+          </div>
+        </form>
+        <div className="widget-form-preview">
+          <Preview image={image} description={description}/>
+        </div>
+      </div>
+    )
+  }
+}
+
+class Section extends Component {
+  render() {
+    const { type, fields, setFieldState, state } = this.props
+
+    return (
+      <div className="widget-form-section">
+        <div className="widget-form-section-type">
+          {type === 'Setup'
+            ? 'Setup — Now set up your widget'
+            : 'Source — Where is your data coming from?'
+          }
+        </div>
+        <div className="widget-form-section-fields">
+          {fields.map(( field, id ) => (
+              <Row
+                {...field}
+                setFieldState={setFieldState}
+                state={state}
+                key={`field-${id}`}
+              />
+            )
+          )}
+        </div>
+      </div>
+    )
+  }
+}
+
+class Preview extends Component {
+  render() {
+    const { description, image, title } = this.props
+    return (
+      <div className="widget-form-preview">
+        <img alt={title} src={image} />
+        <div>{description}</div>
+      </div>
+    )
+  }
+}
+
+class Field extends Component {
+  render() {
+    const { type } = this.props
+
+    if( type === 'text' ) {
+      return <Text {...this.props} />
+    } else if( type === 'dropdown' ) {
+      return <Dropdown {...this.props} />
+    } else if( type === 'nested-dropdown' ) {
+      return <NestedDropdownArea {...this.props} />
+    } else if( type === 'radio' ) {
+      return <RadioField {...this.props} />
+    } else {
+      return <div>Not a valid component type >_<</div>
```

Comment:
It would be instead of the greater than `>` and less than `<` symbols - some browsers will not render correctly if you have those symbols because they are interpreted as the end and beginning of an html tag.  A `>` should be replaced by `&gt;` and a `<` should be replaced by a `&lt;`.  See https://developer.mozilla.org/en-US/docs/Glossary/Entity for more info.